File: toki/belajar/dasar/P24BilanganAgakPrima.cpp
#include <bits/stdc++.h>
using namespace std;

int AP(int x)
{
	int bts = pow(x,0.5)+1, count=3;
	for(int i=2; i<bts; i++)
	{
		if(x%i == 0)
		{
			count +=1;
			if(count>4) return false;
		}
	}
	return true;
}

int main()
{
	int n,x;
	cin>>n;
	for(int i=0; i<n; i++)
	{
		cin>>x;
		if(AP(x)) cout<<"YA\n";
		else cout<<"BUKAN\n";
	}
	return 0;
}

File: toki/belajar/dasar/P24BilanganAgakPrima.py
# P24BilanganAgakPrima.py : program utk mengecek suatu bilangan agak prima.
# Ridhan Fadhilah, 03-02-2016

def AP(x):
    bts = int(x**0.5)
    count = 0
    for i in range(2,bts+1):
        if x % i == 0:
            count += 1
            if count > 1:
                return False
    return True

n = int(input())
for i in range(n):
    if AP(int(input())):
        print('YA')
    else:
        print('BUKAN')
